controllers/operatorpolicy_status.go

```diff
@@ -451,3 +523,82 @@ func opGroupTooManyObjs(opGroups []unstructured.Unstructured) []policyv1.Related
 
 	return objs
 }
+
+func missingCSVObj(sub *operatorv1alpha1.Subscription) policyv1.RelatedObject {
+	return policyv1.RelatedObject{
+		Object: policyv1.ObjectResource{
+			Kind:       clusterServiceVersionGVK.Kind,
+			APIVersion: clusterServiceVersionGVK.GroupVersion().String(),
+			Metadata: policyv1.ObjectMetadata{
+				Name:      sub.Status.CurrentCSV,
+				Namespace: sub.GetNamespace(),
+			},
+		},
+		Compliant: string(policyv1.NonCompliant),
+		Reason:    reasonWantFoundDNE,
+	}
+}
+
+func existingCSVObj(csv *operatorv1alpha1.ClusterServiceVersion) policyv1.RelatedObject {
+	compliance := policyv1.NonCompliant
+	if csv.Status.Phase == operatorv1alpha1.CSVPhaseSucceeded {
+		compliance = policyv1.Compliant
+	}
+
+	return policyv1.RelatedObject{
+		Object: policyv1.ObjectResource{
+			Kind:       clusterServiceVersionGVK.Kind,
+			APIVersion: clusterServiceVersionGVK.GroupVersion().String(),
+			Metadata: policyv1.ObjectMetadata{
+				Name:      csv.Name,
+				Namespace: csv.GetNamespace(),
+			},
+		},
+		Compliant: string(compliance),
+		Reason:    string(csv.Status.Reason) + "Phase" + string(csv.Status.Phase),
+		Properties: &policyv1.ObjectProperties{
+			UID: string(csv.GetUID()),
+		},
+	}
+}
+
+func missingDeploymentObj(name string, namespace string) policyv1.RelatedObject {
+	return policyv1.RelatedObject{
+		Object: policyv1.ObjectResource{
+			Kind:       deploymentGVK.Kind,
+			APIVersion: deploymentGVK.GroupVersion().String(),
+			Metadata: policyv1.ObjectMetadata{
+				Name:      name,
+				Namespace: namespace,
+			},
+		},
+		Compliant: string(policyv1.NonCompliant),
+		Reason:    reasonWantFoundDNE,
+	}
+}
+
+func existingDeploymentObj(dep *appsv1.Deployment) policyv1.RelatedObject {
+	compliance := policyv1.NonCompliant
+	reason := "Deployment Unavailable"
+
+	if dep.Status.UnavailableReplicas == 0 {
+		compliance = policyv1.Compliant
+		reason = "Deployment Available"
+	}
+
+	return policyv1.RelatedObject{
+		Object: policyv1.ObjectResource{
+			Kind:       deploymentGVK.Kind,
+			APIVersion: deploymentGVK.GroupVersion().String(),
+			Metadata: policyv1.ObjectMetadata{
+				Name:      dep.Name,
+				Namespace: dep.Namespace,
+			},
+		},
```

Comment:
```suggestion
		Object: policyv1.ObjectResourceFromObj(dep),
```
Unless the kind information is missing from the object, sometimes that happens 😅